jscomp/test/gpr_4274_test.js

```diff
@@ -5,12 +5,14 @@ var Belt_Array = require("../../lib/js/belt_Array.js");
 
 var N = {};
 
-function f(X, xs) {
-  X.forEach(xs, {
-        i: (function (x) {
-            console.log(x.x);
-          })
-      });
+function f(X) {
```

Comment:
This seems like our only regression, at least in what respects tests having caught them.

There's a difference in the lambda.

### 4.12

```lisp
       (function X/148 xs/150 void
         (let (X/149 = X/148)
           (#full_apply
             (apply (opaque (field:forEach/0 X/149)) xs/150
               (makeblock 0[i] (function x/256 void (log (field:x/0 x/256))))))))

```

### 4.14

```lisp
       (function X/355
         (let (X/356 = X/355)
           (function xs/357 void : int
             (#full_apply
               (apply (opaque (field:forEach/0 X/356)) xs/357
                 (makeblock 0[i]
                   (function x/361 void : int (log (field:x/0 x/361)))))))))
```

Reply:
It seems like everything still works. I'll add a test demonstrating that and we can  move on with this.